### PR TITLE
Properly spreads properties in Tab

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Fixes bug where props were being overwritten rather than cloned in Tab

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/tabs/src/Tabs.tsx [hunk 1]
```diff
@@ -234,6 +234,7 @@ function Tabs({
     const { disabled, onClick } = child.props;
 
     const tabProps = {
+      ...child.props,
       disabled,
```
Comment (lazytype) [inline packages/tabs/src/Tabs.tsx:238]: we can remove `disabled` since it's already in `child.props`

File: packages/tabs/src/Tabs.tsx [hunk 2]
```diff
@@ -234,6 +234,7 @@ function Tabs({
     const { disabled, onClick } = child.props;
 
     const tabProps = {
+      ...child.props,
```
Comment (lazytype) [inline packages/tabs/src/Tabs.tsx:237]: a few things:
- we should preserve `child.props.onKeyDown`
- we should preserve `child.props.className`
- spread should go last so the child can overwrite `as` and `darkMode` if they wish. not sure whether it makes sense for `disabled` to be overwritable

File: packages/tabs/src/Tabs.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/tabs/src/Tabs.tsx:238]: 👍

File: packages/tabs/src/Tabs.tsx [hunk 3]
```diff
@@ -231,25 +231,31 @@ function Tabs({
...
 
     const isTabSelected = index === selected;
-    const { disabled, onClick } = child.props;
+    const { disabled, onClick, onKeyDown, className, ...rest } = child.props;
 
     const tabProps = {
-      ...child.props,
-      disabled,
       as,
+      disabled,
```
Comment (lazytype) [inline packages/tabs/src/Tabs.tsx:238]:
```suggestion
```

File: packages/tabs/src/Tabs.tsx [hunk 4]
```diff
@@ -231,25 +231,31 @@ function Tabs({
...
+      className: cx(
+        {
+          [modeColors[mode].activeStyle]: isTabSelected,
+          [cx(modeColors[mode].disabledColor, disabledStyle)]: disabled,
+        },
+        className,
+      ),
+      onKeyDown: (event: KeyboardEvent) => {
+        handleArrowKeyPress(event);
+        onKeyDown?.(event);
```
Comment (lazytype) [inline packages/tabs/src/Tabs.tsx:250]: just to be a bit more consistent with the ordering of the callbacks
```suggestion
        onKeyDown?.(event);
        handleArrowKeyPress(event);
```

File: packages/tabs/src/Tabs.tsx [hunk 5]
```diff
@@ -231,25 +231,31 @@ function Tabs({
     }
 
     const isTabSelected = index === selected;
-    const { disabled, onClick } = child.props;
+    const { disabled, onClick, onKeyDown, className, ...rest } = child.props;
```
Comment (lazytype) [inline packages/tabs/src/Tabs.tsx:234]:
```suggestion
    const { onClick, onKeyDown, className, ...rest } = child.props;
```

**Review by lazytype — APPROVED** [review]

File: packages/tabs/src/Tabs.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/tabs/src/Tabs.tsx:238]: I pulled out disabled from props and am now only spreading rest so if I don't pass it explicitly then it won't be passed

File: packages/tabs/src/Tabs.tsx (hunk 5 above)
Comment (brookescarlett) [reply packages/tabs/src/Tabs.tsx:234]: We use the disabled value a couple of times (in className concat and onClick)

Comment (brookescarlett) [reply packages/tabs/src/Tabs.tsx:234]: If you feel strongly though, happy to make these changes

Comment (lazytype) [reply packages/tabs/src/Tabs.tsx:234]: no worries

**Pull request merged**